adobe/spectrum-web-components
# fix(dialog): fade-out animation when lazy loaded on popover overlays

## Description

This PR updates the moment of the `close` event dispatch from the `DialogBase` to happen after all the component is done animating. If the transition duration is 0, the event will dispatch instantly as before.

## Related issue(s)

- N/A

## Motivation and context

Using a `DialogBase` inside a lazy-loaded overlay content would not always allow for the fade-out animations to complete. Unsure why this issue happens only with the popover overlay, but this fix seems more in line with the documentation stating that [the close event "announces that the dialog has been closed"](https://opensource.adobe.com/spectrum-web-components/components/dialog-base/api/#:~:text=close,has%20been%20closed.).

## How has this been tested?




    1. Go to the [lazy loaded DialogBase storybook](https://imizga-trigger-directive-transition--spectrum-web-components.netlify.app/storybook/?path=/story/dialog-base--lazy-loaded)
    2. Click on the button to open the dialog
    3. Click the button inside the dialog to close it
    4. Observe that the animation happens smoothly
    5. Repeat 

    1. Go to the [lazy loaded DialogBase storybook](https://imizga-trigger-directive-transition--spectrum-web-components.netlify.app/storybook/?path=/story/dialog-base--lazy-loaded)
    2. Activate the "Reduce Motion" switch from the storybook decorator
    3. Click on the button to open the dialog
    4. Click the button inside the dialog to close it
    6. Observe that the the dialog closes instantly
    7. Repeat


## Screenshots (if appropriate)

Before:

https://github.com/user-attachments/assets/6b25d1bd-f382-4fe0-86b3-d36c48e295b1

After:

https://github.com/user-attachments/assets/cdddf104-eb57-48f7-ab5f-8aeb612b41f6


## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
